Amplify/Categories/DataStore/Model/Collection/List+Model.swift

```diff
@@ -6,82 +6,97 @@
 //
 
 import Foundation
+import Combine
 
-/// `List<ModelType>` is a DataStore-aware custom `Collection` that is capable of loading
-/// records from the `DataStore` on-demand. This is specially useful when dealing with
+/// `List<ModelType>` is a custom `Collection` that is capable of loading
+/// records from a data source. This is specially useful when dealing with
 /// Model associations that need to be lazy loaded.
-///
-/// When using `DataStore.query(_ modelType:)` some models might contain associations
-/// with other models and those aren't fetched automatically. This collection keeps track
-/// of the associated `id` and `field` and fetches the associated data on demand.
 public class List<ModelType: Model>: Collection, Codable, ExpressibleByArrayLiteral {
-
     public typealias Index = Int
     public typealias Element = ModelType
     public typealias Elements = [Element]
```

Comment:
nit: Can we remove this if it's not required for protocol conformance? I don't think it really adds anything to understanding, unless we plan on changing the underlying storage of elements at some point